gunrock/framework/problem.hxx

```diff
@@ -22,22 +22,30 @@ namespace gunrock {
  *
  * @tparam graph_t
  */
-template <typename graph_t>
+template <typename graph_t, typename param_t, typename result_t>
 struct problem_t {
   using vertex_t = typename graph_t::vertex_type;
   using edge_t = typename graph_t::edge_type;
   using weight_t = typename graph_t::weight_type;
 
   graph_t graph_slice;
+  param_t param;
+  result_t result;
   std::shared_ptr<cuda::multi_context_t> context;
 
   auto get_graph() { return graph_slice; }
-
+  
+  void init() {};
+  void reset() {};
+  
```

Comment:
Define null `init` and `reset` default methods -- sometimes a user application doesn't have both of these defined.

Reply:
Must be virtual functions then, user can extend them if they want:
```cpp
  void void init() {};
  virtual void reset() {};
```

If we need to _force_ them to write these, they need to be _pure_ virtual functions;
```cpp
  void void init() = 0;
  virtual void reset() = 0;
```